### DISCUSSION TITLE
Port Forwarding / DMZ

### DISCUSSION THREAD

**Discussion by ArthurZey** [Mygod/VPNHotspot#303] [Ideas]
A particular use case that I imagine might be common is using this as part of a home internet setup. Because my regular router is doing its own DHCP for the LAN, my particular desire would be to put it in the DMZ of the phone's tethered LAN; ie, for all inbound traffic on all ports on the phone's WAN/VPN connection to be forwarded to a particular (in my case, singular) tethered client (ie, my router's WAN connection). That would allow me to continue doing per-port forwarding on my router to clients on its LAN.

The broader functionality would be to support per-port forwarding more granularly.

(I'm very new to this product, so please forgive me if my question is naive!)

Thoughts?

**Comment by Mygod** [comment]
Per-port forwarding could be realized by installing any port forwarding app (which might not need root). Does that achieve what you want?

**Reply by ArthurZey** [reply]
I'm not familiar with such apps, since this is the first time I'm investigating this use case.

Would you happen to be able to recommend one you believe is reputable? (I do have root, for what it's worth.)

To make sure I was being clear, what I'm looking for is that all inbound traffic on my VPN's WAN IP/connection (or alternatively, everything from the phone's WAN in general) is forwarded to a specified tethered client on the phone's 192.168.x.y LAN.

Thank you so much for your speedy reply!

**Reply by Mygod** [reply]
A search on Google Play returns this app: https://play.google.com/store/apps/details?id=com.elixsr.portforwarder

I have tried this but I cannot say whether it is reputable. Also without root, you probably will not be able to set up port forwarding for ports < 1024. Nevertheless I recommend using non-root solutions to prevent port forwarding intervening with other system functionalities.

I do not have plans for implementing full port forwarding/DMZ in this app for now.

**Reply by ArthurZey** [reply]
I'll give it a shot; thanks! I vaguely recall some discussion about static IP reservations for the tethering as I was researching and going through documentation/issues, which I guess I might need here (without fully having tested stuff). Can you point me in the right direction on that?

**Reply by Mygod** [reply]
It is not supported yet: #193
